### PR TITLE
compare candidate query speed

### PR DESCRIPTION
**Pull request by harrisonpim**
Adds scripts to:
- fetch a set of real unique search terms from the reporting cluster
- use those search terms to compare speeds for different indices and queries

Example:
```
➜ yarn compareQuerySpeed

✔ Which namespace do you want to test in? › works
Found 2 indices for testing: works-candidate,works-2021-09-27
✔ Which query should be used with works-candidate? › candidate
✔ Which query should be used with works-2021-09-27? › production
Loading real search terms for querying
Running queries

works-candidate
mean: 2946.123856009762
      0-250 |                                                              | 2
    250-500 | #                                                            | 7
    500-750 |                                                              | 2
   750-1000 |                                                              | 4
  1000-1250 | #                                                            | 5
  1250-1500 |                                                              | 1
  1500-1750 | #                                                            | 12
  1750-2000 | #########                                                    | 78
  2000-2250 | ####                                                         | 31
  2250-2500 | ###################                                          | 154
  2500-2750 | ##############################################               | 378
  2750-3000 | ############################################################ | 495
  3000-3250 | ###########################                                  | 223
  3250-3500 | ################                                             | 129
  3500-3750 | ########                                                     | 66
  3750-4000 | ######                                                       | 49
  4000-4250 |                                                              | 3

works-2021-09-27
mean: 2966.338010982306
      0-250 |                                                              | 1
    250-500 |                                                              | 0
    500-750 |                                                              | 0
   750-1000 | #                                                            | 5
  1000-1250 |                                                              | 1
  1250-1500 |                                                              | 1
  1500-1750 | #                                                            | 8
  1750-2000 | ###########                                                  | 76
  2000-2250 | #####                                                        | 37
  2250-2500 | #################                                            | 117
  2500-2750 | ########################################################     | 380
  2750-3000 | ############################################################ | 404
  3000-3250 | ################################################             | 321
  3250-3500 | ####################                                         | 132
  3500-3750 | ##############                                               | 97
  3750-4000 | ########                                                     | 51
  4000-4250 | #                                                            | 8
```

closes https://github.com/wellcomecollection/platform/issues/5347

### CODE REVIEW

File: rank/services/elasticsearch.ts [hunk 1]
```diff
@@ -25,6 +28,22 @@ export function getRankClient(): Client {
   return rankClient
 }
 
+let reportingClient
+export function getReportingClient(): Client {
+  if (!rankClient) {
```
[the remark this answers is not in the document]
Comment (harrisonpim) [reply rank/services/elasticsearch.ts:33]: oop

[the remark this answers is not in the document]
Comment (harrisonpim) [reply rank/services/elasticsearch.ts:33]: luv 2 copypaste

File: rank/scripts/compareQuerySpeed.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      const { body: searchResp } = await client.msearch({ body })
+      return searchResp.responses.map((r) => r.took)
+    })
+  )
+
+  testIndices.map((index, i) => {
+    const times = multiGetTimes.map((x) => x[i])
+    info(index)
+    info(`mean: ${times.reduce((a, b) => a + b, 0) / times.length}`)
+    console.log(bars(histogram(times)))
```
Comment (jamesgorrie) [inline rank/scripts/compareQuerySpeed.ts:78]: Could we add a passed / failed state here for CI?

**Review by jamesgorrie — COMMENTED** [review]
Could we allow properties that are collected via `prompts` to be passed through via args too?

The reason being that we can then keep running these tests in CI e.g. `yarn compareQuerySpeed --index blah --query bloop`.

File: rank/scripts/compareQuerySpeed.ts (hunk 2 above)
Comment (harrisonpim) [reply rank/scripts/compareQuerySpeed.ts:78]: This script isn't being run in CI yet - I'd like to get to that point soon, but I think adding a human-usable version should be the priority.

Are you alright with me adding that pass/fail state in a subsequent PR?

**Comment by harrisonpim** [comment]
I also really like the idea of making the prompts available as args, but i think that's a separate piece of work which should apply to all the scripts.

**Review by jamesgorrie — APPROVED** [review]

**Pull request merged**